pkg/operator/controllers/guardrails/guardrails_policy.go

```diff
@@ -0,0 +1,171 @@
+package guardrails
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"bytes"
+	"context"
+	"embed"
+	"errors"
+	"fmt"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"text/template"
+	"time"
+
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	kruntime "k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+
+	arov1alpha1 "github.com/Azure/ARO-RP/pkg/operator/apis/aro.openshift.io/v1alpha1"
+	"github.com/Azure/ARO-RP/pkg/operator/controllers/guardrails/config"
+	"github.com/Azure/ARO-RP/pkg/util/dynamichelper"
+)
+
+func (r *Reconciler) getPolicyConfig(ctx context.Context, na string) (string, string, error) {
+	parts := strings.Split(na, ".")
+	if len(parts) < 1 {
+		return "", "", errors.New("unrecognised name: " + na)
+	}
+	name := parts[0]
+
+	instance := &arov1alpha1.Cluster{}
+	err := r.client.Get(ctx, types.NamespacedName{Name: arov1alpha1.SingletonClusterName}, instance)
```

Comment:
instead of fetching the policy config in this function (which is inside a loop), could we move it outside and pass it to the function?